app/src/main/java/com/concordia/mcga/helperClasses/OutdoorDirection.java

```diff
@@ -0,0 +1,176 @@
+    private final int walkingPathColor = 0x801767e8; // transparent blue
+    private LatLng origin, destination;
+    private List<Polyline> polylines;
+    private List<Step> steps;
+    private List<String> instructions;
+    private Leg leg;
+    private GoogleMap map;
+    private Context context;
+    private String transportMode = TransportMode.BICYCLING;
+
+    public OutdoorDirection() {
+        polylines = new ArrayList<>();
+        steps = new ArrayList<>();
+        instructions = new ArrayList<>();
+        transportMode = TransportMode.TRANSIT; // default transport mode
+    }
+
+    /**
+     * Upon successful response from Google Direction Server, Create a path between the origin and destion
+     *
+     * @param direction raw directions in JSON format are wrapped with the Direction class for ease of use
+     * @param rawBody   raw directions in JSON format
+     */
+    @Override
+    public void onDirectionSuccess(Direction direction, String rawBody) {
+        if (direction.isOK()) {
+            Route route = direction.getRouteList().get(0);
+            leg = route.getLegList().get(0);
+            steps = leg.getStepList();
+        }
+    }
+
+    @Override
+    public void onDirectionFailure(Throwable t) {
+        try {
+            throw new MCGAPathFindingException(t.getMessage());
+        } catch (MCGAPathFindingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Makes a https request to get a direction from origin to destination with a specified transport mode.
+     */
+    public void requestDirection() {
+        GoogleDirection.withServerKey(serverKey)
+                .from(origin)
+                .to(destination)
+                .transportMode(transportMode)
+                .unit(Unit.METRIC)
+                .execute(this);
+    }
+
+
+    public void drawPath() {
+        ArrayList<PolylineOptions> polylineOptionList = DirectionConverter.createTransitPolyline(
+                context,
+                steps,
+                transitPathWidth,
+                transitPathColor,
+                walkingPathWidth,
+                walkingPathColor);
+        for (PolylineOptions polylineOption : polylineOptionList) {
+            polylines.add(map.addPolyline(polylineOption));
+        }
+    }
+
+    /**
+     * Deletes Markers and polylines.
+     * sets origin and destionation to null
+     */
+    public void deleteDirection() {
+        origin = null;
+        destination = null;
+        if (polylines != null) {
+            for (Polyline polyline : polylines) {
+                polyline.remove();
+            }
+        }
+    }
+
+    /**
+     * @return list of instructions to get from origin to destination
+     */
+    public List<String> getInstructions() {
+        instructions.clear();
+        for (Step step : steps) {
+            instructions.add(step.getHtmlInstruction());
+        }
+        return instructions;
```

Comment:
Good catch, this line was left from my previous implementation where i was reusing a single OutdoorDirection object and clear was needed then.